ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -582,6 +590,49 @@ object Queries {
     (allTpids diff grouped.keySet).view.map((_, Map.empty[Party, Off])).toMap ++ grouped
   }
 
+  // invariant: each element x of result has `x.values.flatten.size <= size`
+  private[dbbackend] def chunkBySetSize[K, V](
```

Comment:
I'll read on but maybe a good ide to do a little unit test for `chunkBySetSize`.

Reply:
There are a lot of tests for `chunkBySetSize`.